### PR TITLE
chore: fix get_default_site function of Enterprise Customer admin page

### PR DESCRIPTION
**Pull request by yuanyuan-git-tech**
https://2u-internal.atlassian.net/browse/ENT-7322
Screenshot 2023-07-12 at 3 27 04 PM
Screenshot 2023-07-12 at 3 27 14 PM
Screenshot 2023-07-13 at 11 34 24 AM
Screenshot 2023-07-12 at 3 27 33 PM
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/openedx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/openedx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/openedx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/openedx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/models.py [hunk 1]
```diff
@@ -157,6 +158,22 @@ def get_default_customer_type():
...
 
 
+def get_default_site():
+    """
+    Get default site id to use when creating a new EnterpriseCustomer model.
+    The default value depending on what environment the person is in.
+    In production, it should be 'courses.edx.org'.
+    In stage it should be 'courses.stage.edx.org'.
+    """
+    ENVIRONMENT = os.getenv('DJANGO_ENVIRONMENT', 'development')
```
Comment (johnnagro) [inline enterprise/models.py:168]: rather than hard-coding the options here, we should add a configuration setting and/or reuse an existing one such as [LMS_BASE](https://github.com/openedx/edx-platform/blob/cb3a0e91e9b3e39ad5817c5e04abc8c1c9ce0f73/cms/envs/common.py#L770)

That way is more flexible - it can run in any environment (devstack, hosted devstack, sandboxes, etc) where the domains might vary widely, as well as any opensource community install.

**Review by johnnagro — CHANGES_REQUESTED** [review]

File: enterprise/migrations/0177_auto_20230712_1925.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('enterprise', '0176_auto_20230712_1751'),
+    ]
+
+    operations = [
```
Comment (alex-sheehan-edx) [inline enterprise/migrations/0177_auto_20230712_1925.py:12]: is there a reason why these changes are in a second migration file?

File: enterprise/models.py [hunk 3]
```diff
@@ -157,6 +157,18 @@ def get_default_customer_type():
     return enterprise_customer_type.id
 
 
+def get_default_site():
```
Comment (alex-sheehan-edx) [inline enterprise/models.py:160]: 👍

File: enterprise/models.py [hunk 4]
```diff
@@ -223,41 +238,44 @@ class Meta:
     )
 
     enable_data_sharing_consent = models.BooleanField(
+        "Activate data sharing consent prompt",
```
Comment (alex-sheehan-edx) [inline enterprise/models.py:241]: it wasn't clear to me that this param is intended as the `verbose_name`, could we add the named param to this and the other fields below that include verbose names?

**Review by alex-sheehan-edx — COMMENTED** [review]
Looks good! I just have a couple quick comments/questions before we merge. Also don't forget to add your change log and version bumb!

**Review by alex-sheehan-edx — APPROVED** [review]
Make sure you add a change log and bump your version before you merge!

**Review by johnnagro — APPROVED** [review]
